### PR TITLE
UIIN-2671: Jest/RTL: Cover HRIDHandlingSettings components with unit tests

### PR DESCRIPTION
**Pull request by UladzislauKutarkin**
https://folio-org.atlassian.net/browse/UIIN-2671
Here we added test cases for `HRIDHandlingSettings`

Screenshot 2024-05-27 at 13 17 03

### CODE REVIEW

File: src/settings/HRIDHandling/HRIDHandlingSettings.test.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    renderComponent();
+    expect(screen.getByText(/ViewMetaData/)).toBeInTheDocument();
+  });
+
+  it('shows confirmation modal on form submission', async () => {
+    renderComponent();
+
+    fireEvent.click(screen.getByText('confirm'));
+
+    await waitFor(() => expect(mockMutator.hridSettings.PUT).toHaveBeenCalled());
```
Comment (mariia-aloshyna) [inline src/settings/HRIDHandling/HRIDHandlingSettings.test.js:55]: There is no check for confirmation modal appearance as test says but rather for doing an API call

File: src/settings/HRIDHandling/HRIDHandlingSettings.test.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  });
+
+  it('validates required fields', async () => {
+    renderComponent();
+
+    fireEvent.click(screen.getByRole('button', { name:'cancel' }));
+
+    await waitFor(() => expect(mockMutator.hridSettings.PUT).not.toHaveBeenCalled());
+  });
```
Comment (mariia-aloshyna) [inline src/settings/HRIDHandling/HRIDHandlingSettings.test.js:64]: I also do not see here checking the validation of some required fields. It is better to do something like focus and blur some fields and check that the validation error appears

File: src/settings/HRIDHandling/HRIDHandlingSettings.test.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+  it('calls mutator on confirmation', async () => {
```
Comment (mariia-aloshyna) [inline src/settings/HRIDHandling/HRIDHandlingSettings.test.js:66]: No async needed here

File: src/settings/HRIDHandling/HRIDHandlingSettings.test.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    fireEvent.click(screen.getByRole('checkbox'));
+
+    prefixInputs.forEach((input) => {
+      fireEvent.change(input, { target: { value: 'new value' } });
+    });
+
+    fireEvent.click(screen.getByText(/saveAndClose/));
+
+    prefixInputs.forEach((input) => {
+      expect(input).toHaveValue('new value');
```
Comment (mariia-aloshyna) [inline src/settings/HRIDHandling/HRIDHandlingSettings.test.js:79]: It is also not quite the thing that the test says to us. You do not check the mutator here, but fields values?

Comment (UladzislauKutarkin) [reply src/settings/HRIDHandling/HRIDHandlingSettings.test.js:104]: I updated test cases, please check it out

Comment (UladzislauKutarkin) [reply src/settings/HRIDHandling/HRIDHandlingSettings.test.js:104]: Also, the main problem of that component that I can't call submit form from that component

File: src/settings/HRIDHandling/HRIDHandlingSettings.test.js (hunk 2 above)
Comment (UladzislauKutarkin) [reply src/settings/HRIDHandling/HRIDHandlingSettings.test.js]: Updated description

**Review by mariia-aloshyna — APPROVED** [review]

**Review by vashjs — APPROVED** [review]

**Pull request merged**